Gems/Atom/RHI/Code/Include/Atom/RHI/Device.h

```diff
@@ -174,6 +174,7 @@ namespace AZ::RHI
         double GetGPUWaitFrameTime();  // returns wait time before start processing buffers
         double GetGPUWaitAvgFrameTime();  // returns per-buffer averag wait time before start processing buffer
         double GetGPUEndMaxFrameTime();  // returns max commit to done bufer time
+        const AZStd::array<FrameCommands, 4>& GetFrameComands();
```

Comment:
I usa m_FrameTimeLock fence to access the data. You cannot just use the getter with a ref o access the data, it is updated async. Please lock, then make a copy to return. It is not effective, but we do it once per frame.

Reply:
Here you deal with processed data, which is a set on non-overlapping segments. I believe we should view the raw data, the original metal buffers timing.
Please add another container to store original begin-end time from CommandBufferCompleted(const void* buffer, double begin, double end). Then get data from that container. Beware that they are not sorted and typically overlaps.

Reply:
Updated both.